### ISSUE TITLE
Padding fails on the left side for Outlook '13 & '16 when using VML backgrounds.

### ISSUE THREAD

**Issue by dcoursey** [hteumeuleu/email-bugs#32]
Outlook '13 and '16 are incapable of rendering left padding when table elements are laid over a VML background. This applies to both `padding` and `padding-left` attributes, while the other 3 render as intended. 

Inversely, both clients render `margin-left` as intended, without supporting the other 3 margin attributes under these circumstances. 

Therefore, we can add a class/id and padding to a td `<td class="padding" style="padding: 40px;">` and use conditional css to add a left margin for Outlook clients greater than or equal to 13. 
```
<!--[if gte mso 13]>
<style type="text/css">
.padding { margin-left: 40px !important; padding-left: 0 !important; }
</style>
<![endif]-->

```
Setting the padding-left attribue to 0 doesn't seem to have any effect in the conditional css.

EoA test for reference.
https://www.emailonacid.com/app/acidtest/jdHM6uwpCz2b2XS5w5UBTzbExFXtYw2cbYd7ylvl2X9di/list

**Comment by gary-mason** [comment]
Hey @darienallen, I've struggled with the same issue when using bulletproof background images for a single table cell. I since found that the 'inset' values on the 'v:textbox' element work as margin in Outlook, the order being left,top,right,bottom.

**Comment by revelt** [comment]
I'm not sure what exactly do you mean by "bulletproof background images" but standard VML backgrounds generated by https://backgrounds.cm/ don't cause inner table cells inside them to lose `padding-left`. Here's a quick test:

![vml](https://user-images.githubusercontent.com/8344688/38910634-2a58e72e-42c2-11e8-8219-b36a4b5afe1d.gif)

Letter is offset from the left using a generous padding of `45px`. If that padding was down, it would not appear in the centre of the table cell.

Here's the HTML code: https://gist.github.com/revelt/2412138ca045fe879d1851ce6b1b461c

Here's how it looks in Outlook:

![screen shot 2018-04-18 at 04 27 24](https://user-images.githubusercontent.com/8344688/38910671-63f2c612-42c2-11e8-83be-827734ec359f.png)
![screen shot 2018-04-18 at 04 27 32](https://user-images.githubusercontent.com/8344688/38910679-67b1dda6-42c2-11e8-9a6d-a6f101fcd53e.png)
![screen shot 2018-04-18 at 04 27 36](https://user-images.githubusercontent.com/8344688/38910681-69f1888c-42c2-11e8-96c6-ed276a526c2d.png)

Please challenge my code and prove me otherwise — but is it a VML background and it is a cell within a table with `padding-left` and **it is working fine**.

From my practice, I've implemented VML backgrounds on many brand email template headers and never had problems with that. There are issues with `line-height` and `mso-line-height-rule:exactly;` thing but not padding inside VML...

**Comment by darienallen** [comment]
Apologies to all, it seems my initial EoA test was deleted accidentally. @gary-mason you are absolutely correct. This bug and solution was submitted on the premise I did not want to use a <v:textbox> to solve for an unreasonable padding issue with basic CSS. 

@codsen I was unclear, which is more so now that my test has been deleted. Left-padding seems to disappear when a table element is nested more than one `<td>` deep, so if you nest another `<table> (and friends)` inside your current padded `<td>`, say if it's being used as a container, I expect it to be completely ignored and all 2nd-child content will be against the left edge.

I'll run a new test by the end of the week to follow up, always appreciate the feedback

**Comment by darienallen** [comment]
@gary-mason @revelt Original post updated to reflect your comments and concerns, 

Happy coding, 
Darien

**Comment by revelt** [comment]
Good stuff, I'll review the HTML

**Comment by jkupczak** [comment]
@darienallen 

How does the "Second alternative option" you mentioned in your update work? I was able to use your first solution, but I don't understand the second solution.

**Comment by larrydunc** [comment]
I haven't been able to get any of the solutions to work. Does any one have any other suggestions to deal with a row of images I need to center on a vml background.

**Comment by rparmentier** [comment]
Here are some late week ramblings after I stumbled on this bug again last week. Here's what I learnt:

- Any table nested in a padded cell in VML will ignore the padded cell left padding and double the right padding (Thanks Nicole Merlin for the tip on that!)
- This only affects table and not semantic elements like h1 or p.
- A working solution is to wrap a table around the child table that isn't correctly padded and apply a `margin-left` of the same value as the desired padding. However this won't solve the fact that the right side has double the padding. And I have no solution for this.

Here's the code example I used (
[attached is the full code](https://github.com/user-attachments/files/19605548/outlook-vml-padding-bug.html.zip)):

```html
<!--[if mso]>
	<v:rect stroked="false" style="position:absolute; z-index:1; left:0;top:0;width:600px;height:300px;" xmlns:v="urn:schemas-microsoft-com:vml">
		<v:fill type="frame" src="https://i.imgur.com/XhspuWJ.jpeg" />
		<v:textbox inset="10px,10px,10px,10px">
<![endif]-->
	<table border="0" cellpadding="0" cellspacing="0" style="width:100%; background-color:#fff;">
		<tr>
			<td style="padding:30px;">
				<h2>Lorem ipsum</h2>
				<p>
					Lorem ipsum, dolor sit, amet consectetur.
				</p>
				<table border="0" cellpadding="0" cellspacing="0" role="presentation" style="background:#231F20;">
					<tr>
						<td style="padding:10px; color:#fff;">
							<p>Lorem ipsum, dolor sit, amet consectetur adipisicing elit.</p>
						</td>
					</tr>
				</table>
				<table border="0" cellpadding="0" cellspacing="0" role="presentation" style="background:#231F20;width:100%;">
					<tr>
						<td style="padding:10px; color:#fff;">
							<p>Lorem ipsum, dolor sit, amet consectetur adipisicing elit.</p>
						</td>
					</tr>
				</table>
				<table border="0" cellpadding="0" cellspacing="0" role="presentation">
					<tr>
						<td style="margin-left:30px;">
							<table border="0" cellpadding="0" cellspacing="0" role="presentation" style="background:#231F20;">
								<tr>
									<td style="padding:10px; color:#fff;">
										<p>Lorem ipsum, dolor sit, amet consectetur adipisicing elit. Lorem ipsum, dolor sit, amet consectetur adipisicing elit. Lorem ipsum, dolor sit, amet consectetur adipisicing elit.</p>
									</td>
								</tr>
							</table>
						</td>
					</tr>
				</table>
			</td>
		</tr>
	</table>
<!--[if mso]>
  </v:textbox>
</v:rect>
<![endif]-->
```

And here's a resulting screenshot from Email on Acid:

![Image](https://github.com/user-attachments/assets/b111e342-ef91-40d6-9130-3732ebec3ad0)

**Comment by matthieuSolente** [comment]
To address this issue, I propose six different, easy-to-implement solutions. Each has its own advantages and disadvantages depending on the scenario.
https://codepen.io/matthieuSolente/pen/PwoLRpy

For this particular issue, it is commonly said that the padding-left is removed and added to the padding-right, doubling the padding-right and forcing the elements to align to the left. However, upon closer inspection, nothing indicates that the right padding is being doubled. On the contrary, in the code, the padding specified on the first cell adjacent to the VML is respected. Then, in the adjacent tables, to which Outlook applies the .MsoNormalTable class, the horizontal padding is completely removed. This class defines the following style for the table: mso-padding-alt:0in 5.4pt 0in 5.4pt;

Interestingly, on versions of Outlook prior to 2013, the first padding specified on the cell applies to all adjacent elements, but from Outlook 2013 onwards, the padding works for the first cell, but the inline padding indicates zero values, and in the style, we see the horizontal padding reduced to 0in.

We’re not entirely sure what’s happening, as the same code works on earlier versions, but not on those later than 2013.

There are also other properties applied to tables, such as "mso-style-noshow:yes." This attribute is generally applied to HTML elements like tables or table cells. When set to "yes," the element is hidden in versions of Outlook that use the Word rendering engine, but remains visible in other email clients.

There are also properties like "mso-yfti-irow," "mso-yfti-firstrow," and "mso-yfti-irow" that are used to identify and number rows in the HTML tables of emails.

"mso-xlrowspan" doesn’t appear here, but just for your information, it’s used to fix display issues related to the HTML rowspan attribute.

The main property of interest is "mso-yfti-tbllook," which is applied to <table> elements to specify styles such as borders, spacing, and other visual properties. This one could be important for us as it always has the value 1184, but there is no documentation available to know what the other possible values are.

We could consider more well-known properties like "mso-table-lspace" and "mso-table-rspace," but they don’t seem to have any effect in solving our problem.

In summary, there is no doubling of right padding, but rather a complete removal of the horizontal padding from Outlook 2013 onwards in nested tables within a VML component. Nothing in the code truly explains what’s going on. The "mso-yfti-tbllook" property seems interesting, but there is very little or no information about it. Properties specific to VML, such as "mso-wrap-distance-left," do not work either.

Please check the solutions on my code pen to solve this issue; : )
